### ISSUE TITLE
Oracle design draft

### ISSUE THREAD

**Issue by master-davidlee** [bandotio/unsecured-lending#3]
Use Offchain-woker to access multiple APIs that provide prices, then implement simple aggregation algorithms to improve the accuracy of the data.

**Comment by master-davidlee** [comment]
![image](https://user-images.githubusercontent.com/49427668/117523613-f1923b80-afeb-11eb-9a49-e4b9f754dbde.png)

**Comment by bandotadmin** [comment]
According to this design, it needs to run a substrate node with ocw pallet, right? And the left side needs to be refined. How to call ocw interfaces with ink! contract?

**Comment by master-davidlee** [comment]
yeah, OCW-pallet needs to be added to runtime,  and the ocw will store the price on-chain, so the ink can get it through chain extension. Specifically, there are several things that need to be done:
- confirm serval API for  prices
- OCW fetch data from API use HTTP and send unsigned_withpayload transaction to store price on-chain
- set ink chain-extension both on OCW-pallet side and ink contract-side
